### PR TITLE
fix: [DVM2.0-audit-code-clean]: Address additional formatting and syling issues from code reading session

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

After an internal code reading session, we came up with a number of changes we'd like to make to the DVM contracts. These are formatting, comment gas and code layout related.

### CODE REVIEW

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 1]
```diff
@@ -702,19 +663,27 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
     function updateTrackersRange(address voterAddress, uint64 maxTraversals) external {
         resolveResolvablePriceRequests();
         _updateAccountSlashingTrackers(voterAddress, maxTraversals);
     }
 
+    /**
+     * @notice Resolves all resolvable price requests. This function traverses all pending price requests and resolves
+     * them if they are resolvable. It also rolls requests, if needed and deletes requests, if required.
+     */
     function resolveResolvablePriceRequests() public {
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:679]: Should we name this function differently?

`processResolvablePriceRequests`?

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 2]
```diff
@@ -765,24 +735,23 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         VoterStake storage voterStake = voterStakes[voterAddress];
         int256 slash = voterStake.unappliedSlash; // Load in any unapplied slashing from the previous iteration.
         uint64 requestIndex = voterStake.nextIndexToProcess; // Traverse all requests from the last considered request.
-        uint64 requestsTraversed = 0; // Tracker to limit the number of requests to traverse in this call.
-        while (requestIndex < resolvedPriceRequestIds.length && requestsTraversed < maxTraversals) {
-            requestsTraversed = unsafe_inc_64(requestsTraversed);
+        // TODO: add a comment explaining how this while loop works
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:738]: Proposal for the comment explaining the while loop:

```
        // The following mechanism serves to incentivize informed and participatory voting. The loop calculates slashing
        // based on voting activity and applies it to a voter. If the voter voted correctly, they are rewarded. If they
        // voted incorrectly or did not vote, they are slashed. The loop continues until the specified number of
        // traversals or the end of the list of resolved requests is reached.
```

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 3]
```diff
@@ -765,24 +735,23 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
             // Calculate aggregate metrics for this round.
             (uint256 wrongVoteSlashPerToken, uint256 noVoteSlashPerToken) =
-                slashingLib.calcSlashing(totalStaked, vote.results.totalVotes, totalCorrectVotes, request.isGovernance);
+                slashingLibrary.calcSlashing(totalStaked, totalVotes, totalCorrectVotes, request.isGovernance);
 
-            // During this round's tracker calculation, we deduct the pending stake from the voter's total stake. Also,
-            // the pending stakes of voters in a given round are excluded from the cumulativeStakeAtRound;
-            // _computePendingStakes handles this. Thus, the voter's stakes during the active reveal phase of this round
-            // won't be included in the slashes calculations.
+            // Use the effective stake as the difference between the current stake and pending stake. They will having
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:753]:
```suggestion
            // Use the effective stake as the difference between the current stake and pending stake. They will have
```

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 4]
```diff
@@ -964,12 +929,11 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
         uint32 currentRoundId = uint32(getCurrentRoundId());
 
-        // Load in the last resolved index for this round. This means
+        // Load in the last resolved index for this round to continue off from where the last caller left.
         uint64 requestIndex = rounds[currentRoundId].resolvedIndex;
-        uint64 requestsTraversed = 0;
-        // Traverse over all pending requests, bounded by maxTraversals.
-        while (requestIndex < pendingPriceRequestsIds.length && requestsTraversed < maxTraversals) {
-            requestsTraversed = unsafe_inc_64(requestsTraversed);
+        //TODO: improve this comment Traverse over all pending requests, bounded by maxTraversals.
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:934]: Proposal:
```
        // Processes all pending price requests, resolving those that are resolvable and deleting or rolling those are 
        // not based on the max number of rolls allowed. It continues until the maximum number of traversals or the end 
        // of the list of pending requests is reached.
```

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol (hunk 1 above)
Comment (chrismaree) [reply packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:672]: sure. done

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 5]
```diff
@@ -738,27 +711,27 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
     // Updates the slashing trackers of a given account based on previous voting activity. This traverses all resolved
     // requests for each voter and for each request checks if the voter voted correctly or not. Based on the voters
     // voting activity the voters balance is updated accordingly. The caller can provide a maxTraversals parameter to
-    // limit the number of resolved requests to traverse in this call. This is useful if the number of resolved requests
-    // is large and the update needs to be split over multiple transactions.
+    // limit the number of resolved requests to traverse in this call to bound the gas used. Note that throughout this
+    //function we use the voters unappliedSlash to track any unapplied slashing still to be applied to the voter.
     function _updateAccountSlashingTrackers(address voterAddress, uint64 maxTraversals) internal {
         VoterStake storage voterStake = voterStakes[voterAddress];
-        int256 slash = voterStake.unappliedSlash; // Load in any unapplied slashing from the previous iteration.
```
Comment (chrismaree) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:759]: we can rather re-use the `unappliedSlash` variable throughout the function. this saves a slot and decreases an extra check at the end.

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 6]
```diff
@@ -466,19 +429,17 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
         address voter = getVoterFromDelegate(msg.sender);
         VoteSubmission storage voteSubmission = voteInstance.voteSubmissions[voter];
 
-        // Can only reveal in the reveal phase.
-        require(getVotePhase() == Phase.Reveal, "Reveal phase has not started yet");
-        // 0 hashes are disallowed in the commit phase, so they indicate a different error.
-        // Cannot reveal an uncommitted or previously revealed hash
+        require(getVotePhase() == Phase.Reveal, "Reveal phase has not started yet"); // Can only reveal in reveal phase.
+
+        // 0 hashes are blocked in commit, so they indicate a different error: voter did not commit or already revealed.
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:442]: Super nit, I think I prefer Zero here if it makes sense:
```suggestion
        // Zero hashes are blocked in commit, so they indicate a different error: voter did not commit or already revealed.
```

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 7]
```diff
@@ -738,27 +711,27 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
         }
     }
 
     // Updates the slashing trackers of a given account based on previous voting activity. This traverses all resolved
     // requests for each voter and for each request checks if the voter voted correctly or not. Based on the voters
     // voting activity the voters balance is updated accordingly. The caller can provide a maxTraversals parameter to
-    // limit the number of resolved requests to traverse in this call. This is useful if the number of resolved requests
-    // is large and the update needs to be split over multiple transactions.
+    // limit the number of resolved requests to traverse in this call to bound the gas used. Note that throughout this
+    //function we use the voters unappliedSlash to track any unapplied slashing still to be applied to the voter.
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:727]:
```suggestion
    // function we use the voters unappliedSlash to track any unapplied slashing still to be applied to the voter.
```

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 8]
```diff
@@ -768,63 +740,54 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
-    // Applies a given slash to a given voter's stake.
-    function _applySlashToVoter(
-        int256 slash,
-        VoterStake storage voterStake,
-        address voterAddress
-    ) internal {
-        if (slash + int256(voterStake.stake) > 0) voterStake.stake = uint256(int256(voterStake.stake) + slash);
+    // Applies a given slash to a given voter's stake. In the event the sum of the slash and the voter's stake is less
+    // than 0, the voter's stake is set to 0 to prevent the voter's stake from going negative. unappliedSlash tracked
+    //all slashing the staker has received but not yet applied to their stake. Apply it then set it to zero.
```
Comment (md0x) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:781]:
```suggestion
    // all slashing the staker has received but not yet applied to their stake. Apply it then set it to zero.
```

**Review by md0x — APPROVED** [review]
LGTM! Nice improvements  ✨

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 9]
```diff
@@ -948,62 +911,63 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
...
-
-            emit RequestResolved(
-                request.lastVotingRound,
-                resolvedPriceRequestIds.length - 1,
-                request.identifier,
-                request.time,
-                request.ancillaryData,
-                resolvedPrice
-            );
+            // If not resolvable, but the round has passed its voting round, then it must be deleted or resolved. First,
```
Comment (Reinis-FRP) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:953]:
```suggestion
            // If not resolvable, but the round has passed its voting round, then it must be deleted or rolled. First,
```

**Review by Reinis-FRP — APPROVED** [review]
looking good!

**Pull request merged**